pyogrio/tests/conftest.py

```diff
@@ -8,7 +8,7 @@
 
 
 _data_dir = Path(__file__).parent.resolve() / "fixtures"
-ALL_EXTS = [".shp", ".gpkg", ".json"]
+ALL_EXTS = [".shp", ".gpkg", ".geojson", ".geojsons", ".fgb"]
```

Comment:
The row reordering indeed only happens for ".fgb" if a spatial index is used, but because ".geojsons" always reorders the rows, the logic needs to be in the tests anyway... so I don't really saw/see the advantage to divert of the default behaviour of ".fgb" (= having a spatial index).

I also centralized the list of drivers that have this behaviour, so it should be easy to maintain:
```
DRIVERS_REORDER_ROWS = {
    "FlatGeobuf",
    "GeoJSONSeq",
}
```

Reply:
Wondering for ".geojsons", I didn't notice that it always reorders rows. For example doing a basic roundtrip with the naturalearth data, it preserved the order:

```
In [5]: df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))

In [6]: pyogrio.write_dataframe(df, "test_world.geojsons", driver="GeoJSONSeq")

In [7]: df_roundtrip = pyogrio.read_dataframe("test_world.geojsons")

In [8]: df_roundtrip
Out[8]: 
       pop_est      continent                      name iso_a3  gdp_md_est                                           geometry
0       920938        Oceania                      Fiji    FJI      8374.0  MULTIPOLYGON (((180.00000 -16.06713, 179.41351...
1     53950935         Africa                  Tanzania    TZA    150600.0  POLYGON ((33.90371 -0.95000, 31.86617 -1.02736...
2       603253         Africa                 W. Sahara    ESH       906.5  POLYGON ((-8.66559 27.65643, -8.81783 27.65643...
3     35623680  North America                    Canada    CAN   1674000.0  MULTIPOLYGON (((-122.84000 49.00000, -120.0000...
4    326625791  North America  United States of America    USA  18560000.0  MULTIPOLYGON (((-122.84000 49.00000, -122.5000...
..         ...            ...                       ...    ...         ...                                                ...
172    7111024         Europe                    Serbia    SRB    101800.0  POLYGON ((18.82982 45.90887, 19.07277 45.52151...
173     642550         Europe                Montenegro    MNE     10610.0  POLYGON ((20.07070 42.58863, 20.25758 42.81275...
174    1895250         Europe                    Kosovo    -99     18490.0  POLYGON ((20.59025 41.85541, 20.71731 41.84711...
175    1218208  North America       Trinidad and Tobago    TTO     43570.0  POLYGON ((-61.68000 10.76000, -61.66000 10.365...
176   13026129         Africa                  S. Sudan    SSD     20880.0  POLYGON ((30.83385 3.50917, 31.24556 3.78190, ...

[177 rows x 6 columns]

In [9]: df
Out[9]: 
       pop_est      continent                      name iso_a3  gdp_md_est                                           geometry
0       920938        Oceania                      Fiji    FJI      8374.0  MULTIPOLYGON (((180.00000 -16.06713, 180.00000...
1     53950935         Africa                  Tanzania    TZA    150600.0  POLYGON ((33.90371 -0.95000, 34.07262 -1.05982...
2       603253         Africa                 W. Sahara    ESH       906.5  POLYGON ((-8.66559 27.65643, -8.66512 27.58948...
3     35623680  North America                    Canada    CAN   1674000.0  MULTIPOLYGON (((-122.84000 49.00000, -122.9742...
4    326625791  North America  United States of America    USA  18560000.0  MULTIPOLYGON (((-122.84000 49.00000, -120.0000...
..         ...            ...                       ...    ...         ...                                                ...
172    7111024         Europe                    Serbia    SRB    101800.0  POLYGON ((18.82982 45.90887, 18.82984 45.90888...
173     642550         Europe                Montenegro    MNE     10610.0  POLYGON ((20.07070 42.58863, 19.80161 42.50009...
174    1895250         Europe                    Kosovo    -99     18490.0  POLYGON ((20.59025 41.85541, 20.52295 42.21787...
175    1218208  North America       Trinidad and Tobago    TTO     43570.0  POLYGON ((-61.68000 10.76000, -61.10500 10.890...
176   13026129         Africa                  S. Sudan    SSD     20880.0  POLYGON ((30.83385 3.50917, 29.95350 4.17370, ...

[177 rows x 6 columns]

In [10]: df["name"].equals(df_roundtrip["name"])
Out[10]: True
```

Reply:
Ah, OK it was mentioned in inline documentation that it did, and apparently I didn't doublecheck...

Reply:
I remember running into this issue, hence the note - which I would take to mean that it _may_ reorder features / vertices rather than _always_ does; otherwise would not have added special handling for `GeoJSONSeq`.  However, if we're no longer seeing this fail on CI due to reordering issues, we can remove this as a special case.

Reply:
Per the conversation above, if we deliberately prevent spatial index on FGB (which we should), and we no longer see reordering issues for GeoJSONSeq, it would be good to remove special handling of these in the tests.